contribs/gmf/src/d3-lidar-profile/lidarprofilePlot.js

```diff
@@ -0,0 +1,320 @@
+    const domainScaledHeight = domainProfileHeight * domainScale;
+    const domainHeightCentroid = (rangeY[1] + rangeY[0]) / 2;
+    sx = d3.scaleLinear()
+      .domain(rangeX)
+      .range([0, width]);
+    sy = d3.scaleLinear()
+      .domain([
+        domainHeightCentroid - domainScaledHeight / 2,
+        domainHeightCentroid + domainScaledHeight / 2])
+      .range([height, 0]);
+  }
+
+  this.options.profileConfig.scaleX = sx;
+  this.options.profileConfig.scaleY = sy;
+
+  const that = this;
+
+  function zoomed() {
+    if (d3.event.sourceEvent && d3.event.sourceEvent.type === 'mousemove') {
+      if (d3.event.sourceEvent.movementX == 0 && d3.event.sourceEvent.movementY == 0) {
+        return;
+      }
+    }
+
+    that.parent_.measure.clearMeasure();
+
+    const tr = d3.event.transform;
+    const svg = d3.select('svg#profileSVG');
+    const xAxis = d3.axisBottom(sx);
+    const yAxis = d3.axisLeft(sy)
+      .tickSize(-width);
+
+    svg.select('.x.axis').call(xAxis.scale(tr.rescaleX(sx)));
+    svg.select('.y.axis').call(yAxis.scale(tr.rescaleY(sy)));
+    ctx.clearRect(0, 0, width, height);
+
+    svg.select('.y.axis').selectAll('g.tick line')
+      .style('opacity', '0.5')
+      .style('stroke', '#b7cff7');
+
+    that.options.profileConfig.scaleX = tr.rescaleX(sx);
+    that.options.profileConfig.scaleY = tr.rescaleY(sy);
+  }
+
+  const zoom = d3.zoom()
+    .scaleExtent([1, 100])
+    .translateExtent([[0, 0], [width, height]])
+    .extent([[0, 0], [width, height]])
+    .on('zoom', zoomed);
+
+  function zoomEnd() {
+    ctx.clearRect(0, 0, width, height);
+    that.parent_.loader.updateData();
+  }
+
+  zoom.on('end', zoomEnd);
+
+  d3.select('svg#profileSVG')
+    .call(zoom)
+    .on('dblclick.zoom', null);
+
+  d3.select('svg#profileSVG').selectAll('*').remove();
+
+  const svg = d3.select('svg#profileSVG')
+    .attr('width', width + margin.left)
+    .attr('height', height + margin.top + margin.bottom);
+
+  d3.select('svg#profileSVG')
+    .on('mousemove', () => {
+      that.pointHighlight(that);
+    });
+
+
+  const xAxis = d3.axisBottom(sx);
+  const yAxis = d3.axisLeft(sy)
+    .tickSize(-width);
+
+  svg.select('.y.axis').selectAll('g.tick line').style('stroke', '#b7cff7');
+
+  svg.append('g')
+    .attr('class', 'y axis')
+    .call(yAxis);
+
+  svg.append('g')
+    .attr('class', 'x axis')
+    .call(xAxis);
+
+  svg.select('.y.axis').attr('transform', `translate(${margin.left}, ${margin.top})`);
+  svg.select('.x.axis').attr('transform', `translate(${margin.left}, ${height + margin.top})`);
+
+  svg.select('.y.axis').selectAll('g.tick line')
+    .style('opacity', '0.5')
+    .style('stroke', '#b7cff7');
+
+  this.options.profileConfig.previousDomainX = sx.domain();
+  this.options.profileConfig.previousDomainY = sy.domain();
+
+};
+
+/**
+ * Update the Openlayers overlay that displays point position and attributes values
+ * @export
+ * @param {gmf.lidarProfile.plot} that scope of the plot class
```

Comment:
Can't be a useful parameter, please remove it and use "this" inside the function